### PR TITLE
refactor!: remove 'remap' and 'terse' options

### PR DESCRIPTION
**Pull request by dundargoc**
remap:
An option that should always be kept on except for when working with old
Vi scripts is of little use to us.

terse:
Having two methods of setting the same option is redundant and can be
potentially confusing. Recommend adding 's' to 'shortmess' instead.

### CODE REVIEW

File: src/nvim/options.lua [hunk 1]
```diff
@@ -1909,13 +1909,6 @@ return {
       redraw={'current_window'},
       defaults={if_true=false}
     },
-    {
-      full_name='remap',
-      short_desc=N_("mappings to work recursively"),
-      type='bool', scope={'global'},
-      varname='p_remap',
-      defaults={if_true=true}
-    },
```
Comment (gpanders) [inline src/nvim/options.lua:1918]: Follow the pattern used for e.g. `'compatible'`: change `varname` to `'p_force_off'` and the descriptino to `"No description"`.

File: src/nvim/options.lua [hunk 2]
```diff
@@ -1911,9 +1911,9 @@ return {
     },
     {
       full_name='remap',
-      short_desc=N_("mappings to work recursively"),
+      short_desc=N_("No description"),
       type='bool', scope={'global'},
-      varname='p_remap',
+      varname='p_remap_off',
```
Comment (gpanders) [inline src/nvim/options.lua:1916]:
```suggestion
      varname='p_force_on',
```

Comment (dundargoc) [reply src/nvim/options.lua:1916]: ?

Comment (dundargoc) [reply src/nvim/options.lua:1916]: Never mind, I cannot read. Thanks.

File: runtime/doc/vim_diff.txt [hunk 3]
```diff
@@ -554,6 +554,7 @@ Options:
   *'shortname'* *'sn'* *'noshortname'* *'nosn'*
   *'swapsync'* *'sws'*
   *'termencoding'* *'tenc'* (Vim 7.4.852 also removed this for Windows)
+  *'terse'* *'noterse'* (Add 's' to *shortmess* instead)
```
Comment (gpanders) [inline runtime/doc/vim_diff.txt:557]: Single quotes is used for options.

```suggestion
  *'terse'* *'noterse'* (Add "s" to 'shortmess' instead)
```

File: runtime/doc/map.txt [hunk 4]
```diff
@@ -796,9 +796,8 @@ command.  For example: >
    :noremap j k
 This will exchange the cursor up and down commands.
 
-With the normal :map command, when the 'remap' option is on, mapping takes
-place until the text is found not to be a part of a {lhs}.  For example, if
-you use: >
+With the normal :map command a mapping takes place until the text is found not
```
Comment (gpanders) [inline runtime/doc/map.txt:799]: Inserted an extra "a"

```suggestion
With the normal :map command mapping takes place until the text is found not
```

**Review by gpanders — APPROVED** [review]

**Review by zeertzjq — APPROVED** [review]
The `'terse'` option isn't even synchronized with `s` flag in `'shortmess'`, so `&terse` cannot even be used to check the current behavior.

**Pull request merged**